examples/linear_deformation.py

```diff
@@ -0,0 +1,1187 @@
+        self._tmp2 = tmp2
+
+    def gradient(self, x, out=None):
+        """Gradient of the compositon according to the chain rule.
+
+        Parameters
+        ----------
+        x : domain element-like
+            Point in which to evaluate the gradient
+        out : domain element, optional
+            Element into which the result is written
+
+        Returns
+        -------
+        out : domain element
+            Result of the gradient calcuation. If ``out`` was given,
+            the returned object is a reference to it.
+        """
+        if out is None:
+            # adj_op = self._right.derivative(x).adjoint
+            # return adj_op(self._left.gradient(self._right(x)))
+            return self._right.derivative(x).adjoint(
+                self._left.gradient(self._right(x)))
+        else:
+            if self._tmp is not None:
+                tmp_op_ran = self._right(x, out=self._tmp)
+            else:
+                tmp_op_ran = self._right(x)
+
+            if self._tmp2 is not None:
+                tmp_dom = self._left.gradient(tmp_op_ran, out=self._tmp2)
+            else:
+                tmp_dom = self._left.gradient(tmp_op_ran)
+
+            self._right.derivative(x).adjoint(tmp_dom, out=out)
+
+
+class DisplacementOperator(Operator):
+
+    """Operator mapping parameters to an inverse displacement for
+    the domain of the target.
+
+    This operator computes for the momenta::
+
+        alpha --> D(alpha)
+
+    where
+
+        D(alpha)(y) = v(y).
+
+    The operator vector field ``v`` depends on the deformation parameters
+    ``alpha_j`` as follows::
+
+        v(y) = sum_j (K(y, y_j) * alpha_j)
+
+    Here, ``K`` is the RKHS kernel matrix and each ``alpha_j`` is an
+    element of ``R^n``, can be seen as the momenta alpha at control point y_j.
+    """
+
+    def __init__(self, par_space, control_points, discr_space, ft_kernel):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        par_space : `ProductSpace` or `Rn`
+            Space of the parameters. For one-dimensional deformations,
+            `Rn` can be used. Otherwise, a `ProductSpace` with ``n``
+            components is expected.
+        control_points : `TensorGrid` or `array-like`
+            The points ``x_j`` controlling the deformation. They can
+            be given either as a tensor grid or as a point array. In
+            the latter case, its shape must be ``(N, n)``, where
+            ``n`` is the dimension of the template space, and ``N``
+            the number of ``alpha_j``, i.e. the size of (each
+            component of) ``par_space``.
+        discr_space : `DiscreteSpace`
+            Space of the image grid of the template.
+        kernel : `callable`
+            Function to determine the kernel at the control points ``K(y_j)``
+            The function must accept a real variable and return a real number.
+        """
```

Comment:
Currently, we have hard-coded the FT-based implementation. We should offer also the matrix-based one by using an `impl` parameter that allows to choose between e.g. `'matrix'` and `'fourier'` or similar. Of course, that would be part of the kernel interpolation rather than this operator once that refactor is done.